pkg/cable/vxlan/vxlan.go

```diff
@@ -304,7 +307,17 @@ func (v *vxlan) ConnectToEndpoint(endpointInfo *natdiscovery.NATEndpointInfo) (s
 		return endpointInfo.UseIP, fmt.Errorf("failed to add remoteIP %q to the forwarding database", remoteIP)
 	}
 
-	err = v.vxlanIface.AddRoute(allowedIPs, remoteVtepIP, v.vxlanIface.vtepIP)
+	cniIface, err := cni.Discover(v.localCluster.Spec.ClusterCIDR[0])
+	var ipAddress net.IP
+	if err == nil {
+		ipAddress = net.ParseIP(cniIface.IPAddress)
+	} else {
+		klog.Errorf("failed to get the CNI interface IP for cluster CIDR %q, hence using "+
+			"the vxlan interface IP %q", v.localCluster.Spec.ClusterCIDR[0], v.vxlanIface.vtepIP)
+		ipAddress = v.vxlanIface.vtepIP
```

Comment:
Normally we don't program any routing rules for the VxLAN CIDR. So if we use vxlanIface.vtepIP here, will it work? Can you please verify this by modifying the code to use the vtepip once?

Reply:
it was using vxlanIface.vtepIP before this PR . It do fail for some use cases like host-networking traffic from g/w node to a pod not g/w.

Reply:
Now using not src ip for the route if CNI interface Ip is missing.